app/Http/Controllers/Legacy/Web/ActionsController.php

```diff
@@ -90,7 +112,7 @@ public function update(Request $request, Action $action)
             'post_type' => 'string|in:photo,voter-reg,text,share-social,phone-call',
             'callpower_campaign_id' => [
                 'required_if:post_type,phone-call',
-                Rule::unique('actions')->ignore($action->id),
+                Rule::unique('actions')->whereNotNull('callpower_campaign_id')->ignore($action->id),
```

Comment:
Could you explain why we need the `->ignore($action->id)` piece of this? Having trouble wrapping my head around it (sorry I know that piece isn't new here!)

Reply:
Yes this was confusing to me too! So we want the validation logic to only allow one CallPower Campaign ID per action. To do that, we use `->ignore($action->id)`.

[Here's](https://laravel.com/docs/5.8/validation#forcing-a-unique-rule-to-ignore-a-given-id) the documentation for it but lmk if it still doesn't make sense!

Reply:
Ah that documentation helped! So a CallPower Campaign ID can only live on one action, am I understanding that?

Reply:
Yes exactly!